include/bluetooth/mesh/main.h

```diff
@@ -425,14 +425,56 @@ int bt_mesh_lpn_set(bool enable);
  */
 int bt_mesh_lpn_poll(void);
 
+/** Low Power Node callback functions. */
+struct bt_mesh_lpn_cb {
+	/** @brief Friendship established.
+	 *
+	 *  This callback notifies the application that friendship has
+	 *  been successfully established.
+	 *
+	 *  @param friend_addr Friend address.
+	 *  @param queue_size  Friend queue size.
+	 *  @param recv_window Low Power Node's listens duration for
+	 *  Friend response.
+	 */
+	void (*established)(uint16_t friend_addr, uint16_t queue_size,
+						uint8_t recv_window);
+
+	/** @brief Friendship terminated.
+	 *
+	 *  This callback notifies the application that friendship has
+	 *  been terminated.
+	 *
+	 *  @param friend_addr Friend address.
+	 */
+	void (*terminated)(uint16_t friend_addr);
+
+	/** @brief Local Poll Request.
+	 *
+	 *  This callback notifies the application that local node has
+	 *  been sended poll request.
+	 *
+	 *  @note This callback will be call before `lpn_cb::established`
+	 *  when attempts to establishment friendship.
+	 *
+	 *  @param retry Retry or first poll request for each transaction.
+	 */
+	void (*polled)(bool retry);
+
+	/** Meant to be used internally, to insert the callback into a list.
+	 * So nobody should mess with it.
+	 */
+	sys_snode_t node;
+};
+
 /** @brief Register a callback for Friendship changes.
  *
  *  Registers a callback that will be called whenever Friendship gets
  *  established or is lost.
  *
  *  @param cb Function to call when the Friendship status changes.
  */
-void bt_mesh_lpn_set_cb(void (*cb)(uint16_t friend_addr, bool established));
+void bt_mesh_lpn_set_cb(struct bt_mesh_lpn_cb *cb);
```

Comment:
Change to `bt_mesh_lpn_register_cb` perhaps, as we can register multiple? Also, documentation text needs an update.